File: Exercices/Séance 1 - C# vs Java-20231230/ActorsDB/src/be/ipl/domaine/PersonList.java
package be.ipl.domaine;

import java.security.InvalidParameterException;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;

public class PersonList {

	private static PersonList instance;
	private Map<String, Person> personMap;

	private PersonList(){
		personMap = new HashMap<String, Person>();
	}

	public static PersonList getInstance(){
		
		if (instance == null)
			instance = new PersonList();
		
		return instance;
	}

	public void addPerson(Person person){
		if (person == null)
			throw new InvalidParameterException();
		personMap.put(person.getName(), person);
	}
	
	public Iterator<Person> personList(){
		return personMap.values().iterator();
	}
	
}

File: Exercices/Séance 1 - C# vs Java-20231230/Ex1/PersonList.cs
namespace be.ipl.domaine;

internal class PersonList
    {

	// ? -> nullable field
	private static PersonList? _instance = null;
	// map -> IDictionnary
	// hashmap -> Dictionnary
		private IDictionary<string, Person> personMap;

		private PersonList()
		{
			personMap = new Dictionary<string, Person>();
		}

		public static PersonList getInstance()
		{

			if (_instance == null)
				_instance = new PersonList();

			return _instance;
		}

		public void AddPerson(Person person)
		{
			if (person == null)
				throw new ArgumentException();
			personMap.Add(person.Name, person);
		}

		public IEnumerator<Person> personList()
		{
			return personMap.Values.GetEnumerator();
		}

	}
